scalanet/test/src/io/iohk/scalanet/peergroup/kademlia/KBucketsSpec.scala

```diff
@@ -0,0 +1,71 @@
+package io.iohk.scalanet.peergroup.kademlia
+
+import io.iohk.scalanet.peergroup.kademlia.Generators._
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+import org.scalatest.prop.GeneratorDrivenPropertyChecks._
+import scodec.bits.BitVector
+
+import scala.util.Random
+
+class KBucketsSpec extends FlatSpec {
+
+  behavior of "KBuckets"
+
+  val kb = new KBuckets(aRandomBitVector())
+
+  they should "retrieve the base node id" in {
```

Comment:
minor comment, in the paper, no kbucket holds the node's self id.

Reply:
Yes, I know. Doing remote lookups, I have to add code to filter out the node's own id (because it is in the kbuckets) but, equally, it is cleaner to return closest node results from the kbuckets knowing the nodes id _is_ there when that id should be part of the result, as that avoids some cumbersome if logic.